src/main/kotlin/org/jitsi/nlj/transform/node/incoming/DuplicateTermination.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright @ 2018 - Present, 8x8 Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.nlj.transform.node.incoming
+
+import java.util.Collections
+import java.util.TreeMap
+import org.jitsi.nlj.PacketInfo
+import org.jitsi.nlj.stats.NodeStatsBlock
+import org.jitsi.nlj.transform.node.TransformerNode
+import org.jitsi.rtp.rtp.RtpPacket
+import org.jitsi.utils.LRUCache
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.logging2.createChildLogger
+
+class DuplicateTermination(parentLogger: Logger) : TransformerNode("Probing termination") {
+    private val logger = createChildLogger(parentLogger)
+    private val replayContexts: MutableMap<Long, MutableSet<Int>> = TreeMap()
+    private var numDuplicatePacketsDropped = 0
+
+    override fun transform(packetInfo: PacketInfo): PacketInfo? {
+        val rtpPacket = packetInfo.packetAs<RtpPacket>()
+        val replayContext = replayContexts.computeIfAbsent(rtpPacket.ssrc) {
+            Collections.newSetFromMap(LRUCache(1500))
```

Comment:
Minor: We now have `LRUCache.lruSet()`.

Minor: I wonder if insertion vs access order makes a difference. Perhaps if senders send the same packte multiple times we want access order to make sure it doesn't fall off the cache (though size 1500 should be plenty)

Reply:
This code is unchanged from how it was before, but both your suggestions sound good.